lib/core/src/lib/info-drawer/info-drawer.component.spec.ts

```diff
@@ -87,25 +87,27 @@ describe('InfoDrawerComponent', () => {
 
 @Component({
     template: `
-    <adf-info-drawer [selectedIndex]="tabIndex" title="Fake Title Custom">
-        <adf-info-drawer-tab label="Tab1">
-        </adf-info-drawer-tab>
-        <adf-info-drawer-tab label="Tab2">
-        </adf-info-drawer-tab>
-        <adf-info-drawer-tab label="Tab3" icon="tab-icon">
-        </adf-info-drawer-tab>
-    </adf-info-drawer>
-       `
+        <adf-info-drawer [selectedIndex]="tabIndex" [icon]="icon" title="Fake Title Custom">
+            <adf-info-drawer-tab label="Tab1">
+            </adf-info-drawer-tab>
+            <adf-info-drawer-tab label="Tab2">
+            </adf-info-drawer-tab>
+            <adf-info-drawer-tab label="Tab3" icon="tab-icon">
+            </adf-info-drawer-tab>
+        </adf-info-drawer>
+    `
 })
 class CustomInfoDrawerComponent extends InfoDrawerComponent {
     tabIndex: number;
+    icon: string;
 }
 
 describe('Custom InfoDrawer', () => {
     let fixture: ComponentFixture<CustomInfoDrawerComponent>;
     let component: CustomInfoDrawerComponent;
     let translateService: TranslateService;
-
+    const nodeIcon = () =>
```

Comment:
This function is repeated twice (it's duplicated in 187th line). I know it is in different describe but you can try to move it at the top, before of first describe (instead of inside describe). This function uses fixture so you can move fixture declaration too before first describe.

Reply:
In that same file we have 3 fixtures for 3 different components, I try to put at the top but due to different fixtures its not getting the proper reference so unit tests are failing.